src/kaleidoscope/driver/keyscanner/ATMegaKeyScannerDescription.h

```diff
@@ -55,12 +55,17 @@
 
 namespace kaleidoscope {
 namespace driver {
-namespace scanner {
+namespace keyscanner {
 
-struct ATMegaScannerDescription : public kaleidoscope::driver::ScannerDescription {
+struct ATMegaKeyScannerDescription : kaleidoscope::driver::KeyScannerDescription {
+  static const uint8_t debounce = 3;
+
+  /*
+   * The following two lines declare an empty array. Both of these must be
+   * shadowed by the descendant keyscanner description class.
+   */
   static constexpr uint8_t matrix_row_pins[] = {};
   static constexpr uint8_t matrix_col_pins[] = {};
```

Comment:
It might be worth adding 
```cpp
static_assert(sizeof(_DeviceDescription::KeyScannerDescription::matrix_row_pins) > 0, "Empty row pins array detected");
// Same for col pins
```
to any part of the code that actually does something with the row pins/col pins arrays.

Reply:
Good idea, added it to `ATMegaKeyScanner::setup()`.